### PR TITLE
[CARBONDATA-2369] Add a document for Non Transactional table with SDK writer guide

### PR DESCRIPTION
**Pull request by ajantha-bhat**
[CARBONDATA-2369] Add a document for Non Transactional table with SDK writer guide
As per PR#2131 [CARBONDATA-2313]

Be sure to do all of the following checklist to help us incorporate 
your contribution quickly and easily:

 - [ ] Any interfaces changed? No
 
 - [ ] Any backward compatibility impacted? No
 
 - [ ] Document update required? yes, updated

 - [ ] Testing done -- NA
       
 - [ ] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA.

### CODE REVIEW

File: docs/sdk-writer-guide.md [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+ writer.write(new String[]{"robot" + (i % 10), String.valueOf(i)});
+ }
+ writer.close();
+ }
+ }
+```
+
+## Datatypes Mapping
+Each of SQL data types are mapped into data types of SDK. Following are the mapping:
+| SQL DataTypes | Mapped SDK DataTypes |
```
[the remark this answers is not in the document]
Comment (ajantha-bhat) [reply docs/sdk-writer-guide.md]: ok. Fixed it.

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/5239/

**Comment by sgururajshetty** [comment]
LGTM

**Comment by ajantha-bhat** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/4190/

File: docs/data-management-on-carbondata.md [hunk 2]
```diff
@@ -174,6 +174,50 @@ This tutorial is going to introduce all commands and data operations on CarbonDa
...
+  
+  sql(s"""
+  |CREATE EXTERNAL TABLE source
+  |STORED BY 'carbondata'
+  |LOCATION '$storeLocation/origin'
+  """.stripMargin)
+  checkAnswer(sql("SELECT count(*) from source"), sql("SELECT count(*) from origin"))
+  ```
+  
+### Create external table on Non-Transactional table data location.
```
[the remark this answers is not in the document]
Comment (ajantha-bhat) [reply docs/data-management-on-carbondata.md]: done. Modified.

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/4195/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/4506/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/5354/

File: docs/sdk-writer-guide.md [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+| DOUBLE | DataTypes.DOUBLE |
+| VARCHAR | DataTypes.STRING |
+| DATE | DataTypes.DATE |
+| TIMESTAMP | DataTypes.TIMESTAMP |
+| STRING | DataTypes.STRING |
+| DECIMAL | DataTypes.createDecimalType(precision, scale) |
+
+
+## API List
+```
```
Comment (gvramana) [inline docs/sdk-writer-guide.md]: Add these methods under class CarbonWriterBuilder

File: docs/sdk-writer-guide.md [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+/**
+* prepares the builder with the schema provided
+* @param schema is instance of Schema
+* @return updated CarbonWriterBuilder
+*/
+public CarbonWriterBuilder withSchema(Schema schema);
```
Comment (gvramana) [inline docs/sdk-writer-guide.md]: 1. Add CarbonWriter class details of methods in classe CarbonWriter
2. Add method buildWriterForCSVInput, buildWriterForAvroInput
3. Add example, writing CSV record and writing Avro record.

File: docs/sdk-writer-guide.md [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+* Sets the output path of the writer builder
+* @param path is the absolute path where output files are written
+* @return updated CarbonWriterBuilder
+*/
+public CarbonWriterBuilder outputPath(String path);
+```
+
+```
+/**
+* If set false, writes the carbondata and carbonindex files in a flat folder structure
```
Comment (gvramana) [inline docs/sdk-writer-guide.md]: What when is the behaviour when set to true, what is the default value for all optional parameters

File: docs/sdk-writer-guide.md [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+*                    If it is null, all dimensions are selected for sorting
+*                    If it is empty array, no columns are sorted
+* @return updated CarbonWriterBuilder
+*/
+public CarbonWriterBuilder sortBy(String[] sortColumns);
+```
+
+```
+/**
+* If set, creates a schema file in metadata folder.
```
Comment (gvramana) [inline docs/sdk-writer-guide.md]: what is the default value, what is the effect of setting isTransactionTable(true/false)

File: docs/sdk-writer-guide.md [hunk 7]
```diff
@@ -0,0 +1,172 @@
...
+* @return updated CarbonWriterBuilder
+*/
+public CarbonWriterBuilder withBlockletSize(int blockletSize);
+```
+
+```
+/**
+* sets the list of columns that needs to be in sorted order
+* @param sortColumns is a string array of columns that needs to be sorted.
+*                    If it is null, all dimensions are selected for sorting
```
Comment (gvramana) [inline docs/sdk-writer-guide.md]: What is the default value if not set. Default value if not set should be mentioned for all APIs

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/4518/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/4213/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/5380/

**Comment by gvramana** [comment]
LGTM

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/4226/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/4532/

**Pull request closed without merging**